spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,331 @@
+    - There are theoretically two cases here:
+        - The dependent resource must be deleted before the parent can be deleted.
+        - Deletion of the parent automatically cascades to all dependent resources. To the best of my knowledge all
+         owning resource relationships in Azure are this kind.
+
+
+Note that sometimes an owning type has its dependent type embedded as a property 
+(for example: `RouteTable` has the property [RouteTablePropertiesFormat](https://schema.management.azure.com/schemas/2020-03-01/Microsoft.Network.json)).
+Most types do not embed the dependent resource directly in the owning resource. We need to cater to both scenarios.
+
+## What does expressing these relationships look like in existing solutions?
+
+This section examines how other operator solutions have tackled these problems. We look at:
+- ARM templates
+- AzureServiceOperator (ASO)
+- k8s-infra
+
+### Related/Linked resources
+
+#### What ARM does
+These are just properties (often but not always called `id`) which refer to the fully qualified ARM ID of another resource. For example see a
+[sample deployment template for a VMSS refering to an existing vnet](https://github.com/Azure/azure-quickstart-templates/blob/master/201-vmss-existing-vnet/azuredeploy.json#L136).
+
+```json
+"properties": {
+  "subnet": {
+    "id": "[resourceId(parameters('existingVnetResourceGroupName'), 'Microsoft.Network/virtualNetworks/subnets', parameters('existingVnetName'), parameters('existingSubNetName'))]"
+  },
+}
+```
+
+#### What ASO does
+Similar to what ARM templates do, ASO uses the decomposition of fully qualified resource id to reference another resource, as seen [here for VMSS -> VNet](https://github.com/Azure/azure-service-operator/blob/92240406aff3863f3a267d8a1dc1e28aa3e841ae/api/v1alpha1/azurevmscaleset_types.go#L25)
+
+```go
+type AzureVMScaleSetSpec struct {
+    ...
+	Location               string `json:"location"`
+	ResourceGroup          string `json:"resourceGroup"`
+	VirtualNetworkName     string `json:"virtualNetworkName"`
+	SubnetName             string `json:"subnetName"`
+    ...
+}
+```
+
+These properties are combined into a fully qualified ARM id like so:
+
+```go
+subnetIDInput := helpers.MakeResourceID(
+    client.SubscriptionID,
+    resourceGroupName,
+    "Microsoft.Network",
+    "virtualNetworks",
+    vnetName,
+    "subnets",
+    subnetName,
+)
+```
+
+Currently ASO does not support cross-subscription references (and some of the resources such as VMSS above don't even support cross-resource group references), but it in theory could by adding some more parameters.
+
+#### What k8s-infra does
+k8s-infra is a bit different in that resource references are in Kubernetes language (namespace + name) and not Azure language (resource group + resource name).
+All resource references are done using a special type `KnownTypeReference` which contains the fully qualified Kubernetes name for the resource.
```

Comment:
Perhaps, consider adding the linkage to the Azure resource, which is the `status.id` field on each resource. This might help to clarify the relationship between the Azure resource and the K8s resource, which is more direct in the aforementioned examples.